src/router.rs

```diff
@@ -133,6 +133,39 @@ impl<T> Router<T> {
     pub fn check_priorities(&self) -> Result<u32, (u32, u32)> {
         self.root.check_priorities()
     }
+
+
+    /// Merge a given router into current one.
+    ///
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use matchit::Router;
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// let mut root = Router::new();
+    /// root.insert("/home", "Welcome!")?;
+    ///
+    /// let mut child = Router::new();
+    /// child.insert("/users/{id}", "A User")?;
+    ///
+    /// root.merge(child)?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn merge(&mut self, other: Self) -> Result<(), InsertError> {
+        let mut result = Ok(());
+        other.root.for_each(|path, value| {
+            match self.insert(path, value) {
+                Ok(..) => true,
+                Err(err) => {
+                    result = Err(err);
+                    false
```

Comment:
Hmm.. this will abort on an error but leave `self` in a partially-merged state. It might make more sense to rollback on error or verify that no routes are conflicting beforehand, guaranteeing that `self` is untouched on error, though this may be a little difficult to implement. Another option is to return a `Vec<InsertError>` of all routes that could not be inserted, which may be simpler. Both seem like valid options as long as they are documented, but an order-dependent partial-merge may be a footgun.